Lib/test/test_ctypes/test_refcounts.py

```diff
@@ -97,5 +97,31 @@ def func(a, b):
         f(1, 2)
         self.assertEqual(sys.getrefcount(ctypes.c_int), a)
 
+    @support.refcount_test
+    def test_callback_py_object_none_return(self):
```

Comment:
AFAICT, this test is rendered pointless in 3.12 due to the adoption of [PEP 683](https://peps.python.org/pep-0683/) (Immortal Objects, Using a Fixed Refcount). Because `None` is immortal, executing `Py_INCREF(Py_None)` and `Py_DECREF(Py_None)` will no longer modify its refcount.

Reply:
Ah, good to know. I assume this isn't implemented yet though, because as of the current state of `main` (447d061bc7b978afedd3b0148715d2153ac726c5), this test still hard-crashes without the fix.